packages/commons/engine_util.js

```diff
@@ -609,6 +611,8 @@ function extractRegExp(doc, expr, opts) {
 
   if (group && match.groups) {
```

Comment:
Hey @InesNi ! Looks good!

Could you please add some comments above the logic explaining the different forks, since we don't have tests for it yet?